src/stp/plays/PenaltyThemPrepare.cpp

```diff
@@ -28,23 +28,23 @@ PenaltyThemPrepare::PenaltyThemPrepare() : Play() {
 uint8_t PenaltyThemPrepare::score(world::World* world) noexcept { return 100; }
 
 void PenaltyThemPrepare::calculateInfoForRoles() noexcept {
-    auto width = field.getFieldWidth();
-    auto length = field.getFieldLength();
+    const auto xPosition = -4 * control_constants::ROBOT_RADIUS;
+    const auto yPosition = Constants::STD_TIMEOUT_TO_TOP() ? field.getFieldWidth() / 2.2 : -field.getFieldWidth() / 2.2;
```

Comment:
What's up with the 2.2? That's what we typically call 'magic values'. It would be cleaner if you would write it like e.g:
double distanceFromCenterLine = field.getFieldWidth()/2.2; so you don't repeat yourself. and then do:
double yPosition = Constants::STD_TIMEOUT_TO_TOP() ? distanceFromCenterLine : -distanceFromCenterLine;
If you want the position to be slightly in the field, then it's probably better to do somethinglike:
field.getFieldWidth() / 2 + robotradius, because that will scale better for various field sizes.

Reply:
I do like that you made these variables const, however!

Reply:
Also, I think auto is a bit convoluted here, I would just use double.